R/UWerr.R

```diff
@@ -344,45 +354,68 @@ uwerrderived <- function(f, data, nrep, S=1.5, pl=FALSE, ...) {
 }
 
 summary.uwerr <- function(uwerr) {
+
   cat("Error analysis with Gamma method\n")
   cat("based on", uwerr$N, "measurements\n")
   if(uwerr$R>1) {
     cat("split in", uwerr$R, "replica with (", uwerr$nrep, ") measurements, respectively\n")
   }
-  cat("The Gamma function was summed up until Wopt=", uwerr$Wopt, "\n\n")
-  cat("value    =", uwerr$value, "\n")
-  cat("dvalue   =", uwerr$dvalue, "\n")
-  cat("ddvalue  =", uwerr$ddvalue, "\n")
-  cat("tauint   =", uwerr$tauint, "\n")
-  cat("dtauint  =", uwerr$dtauint, "\n")
-  if(uwerr$R>1) {
-    cat("Qval     =", uwerr$Qval, "\n")
+  if(uwerr$primary == 1) {
+    cat("The Gamma function was summed up until Wopt=", uwerr$Wopt, "\n\n")
+    cat("value    =", uwerr$value, "\n")
+    cat("dvalue   =", uwerr$dvalue, "\n")
+    cat("ddvalue  =", uwerr$ddvalue, "\n")
+    cat("tauint   =", uwerr$tauint, "\n")
+    cat("dtauint  =", uwerr$dtauint, "\n")
+    if(uwerr$R>1) {
+      cat("Qval     =", uwerr$Qval, "\n")
+    }
+  }
+  else {
+    cat("The Gamma function was summed up until Wopt=", unlist(uwerr$Wopt), "for the ", length(uwerr$Wopt), "observable(s)\n\n")
+    print(uwerr$res) 
   }
 }
 
-plot.uwerr <- function(x, ..., main="x", x11=TRUE, plot.hist=TRUE) {
+plot.uwerr <- function(x, ..., main="x", x11=TRUE, plot.hist=TRUE, index=1, Lambda=100) {
   if(x$primary && plot.hist) {
     if(x11) if(interactive() && (grepl(pattern="X11", x=names(dev.cur()), ignore.case=TRUE) || grepl(pattern="null", x=names(dev.cur()), ignore.case=TRUE))) {
               X11()
             }
     hist(x$data, main = paste("Histogram of" , main))
   }
   if(!is.null(x$Gamma)) {
-    GammaFbb <- x$Gamma/x$Gamma[1]
-    Gamma.err <- gammaerror(Gamma=GammaFbb, N=x$N , W=x$Wmax, Lambda=100)
+    GammaFbb <- NULL
+    Gamma.err <- NULL
+    Wopt <- numeric()
+    Wmax <- numeric()
+    if(x$primary == 1) {
+      GammaFbb <- x$Gamma/x$Gamma[1]
+      Gamma.err <- gammaerror(Gamma=GammaFbb, N=x$N , W=x$Wmax, Lambda=Lambda)
+      Wopt <- x$Wopt
+      Wmax <- x$Wmax
+    }
+    else {
+      GammaFbb <- x$Gamma[[index]]/x$Gamma[[index]][1]
+      Gamma.err <- gammaerror(Gamma=GammaFbb, N=x$N , W=x$Wmax[[index]], Lambda=Lambda)
+      Wopt <- x$Wopt[[index]]
+      Wmax <- x$Wmax[[index]]
+    }
     if(x11) if(interactive() && (grepl(pattern="X11", x=names(dev.cur()), ignore.case=TRUE) || grepl(pattern="null", x=names(dev.cur()), ignore.case=TRUE))) {
               X11()
             }
-    plotwitherror(c(0:x$Wmax),GammaFbb[1:(x$Wmax+1)],
-                  Gamma.err[1:(x$Wmax+1)], ylab="Gamma(t)", xlab="t", main=main)
-    abline(v=x$Wopt+1)
+    plotwitherror(c(0:Wmax),GammaFbb[1:(Wmax+1)],
```

Comment:
could we please use named arguments whenever possible? I think we need to make additional modifications to ```plotwitherror``` in the coming months and it would be good if stuff didn't break as a result...